### PR TITLE
fix(TreeView): raise DragItemsCompleted when dragging items is completed

### LINKED ISSUE #14498: [TreeView] When Dragging and Dropping the TreeViewItem, TreeView.DragItemsCompleted is not called in `.NET 8` WASM
### Current behavior

When Dragging and Dropping the `TreeViewItem`, `TreeView.DragItemsCompleted` is not called in .NET 8 WASM, but it works in previous version.
[1.zip](https://github.com/unoplatform/uno/files/13428550/1.zip)

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

```
<Project ToolsVersion="15.0">
  <ItemGroup>
    <PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="8.0.0-rc.2.23479.6" />
    <PackageVersion Include="Microsoft.Windows.Compatibility" Version="8.0.0-rc.2.23479.10" />
    <PackageVersion Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />
    <PackageVersion Include="Microsoft.WindowsAppSDK" Version="1.4.231008000" />
    <PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
    <PackageVersion Include="Uno.Extensions.Logging.WebAssembly.Console" Version="1.7.0" />
    <PackageVersion Include="Uno.Resizetizer" Version="1.2.1" />
    <PackageVersion Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="5.0.48" />
    <PackageVersion Include="Uno.UniversalImageLoader" Version="1.9.36" />
    <PackageVersion Include="Uno.Wasm.Bootstrap" Version="8.0.4" />
    <PackageVersion Include="Uno.Wasm.Bootstrap.DevServer" Version="8.0.4" />
    <PackageVersion Include="Uno.Wasm.Bootstrap.Server" Version="8.0.4" />
    <PackageVersion Include="Uno.WinUI" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.Lottie" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.DevServer" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.WebAssembly" Version="5.0.48" />
  </ItemGroup>
</Project>

```

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022

### IDE version

17.8

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by TopProgrammer77**
GitHub Issue (If applicable): closes #14498 

## PR Type

What kind of change does this PR introduce?

- Bugfix
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at cd57369</samp>

Added a UI test and improved the drag and drop functionality of the tree view control. Fixed the web assembly port number for the SamplesApp project. Added UI elements to toggle the visibility of a text box that displays drag and drop events.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by dr1rrb — COMMENTED** [review]
Hey thanks for the PR @TopProgrammer77 👍

While the changes makes 100% sense, I'm not sure how it could fix the issue https://github.com/unoplatform/uno/issues/14498. Could add some comments to explain the link?

cc. @MartinZikmund

File: src/Uno.UI.FluentTheme.v2/themeresources_v2.xaml [hunk 1]
```diff
@@ -19837,11 +19837,11 @@
     <Setter Property="HorizontalContentAlignment" Value="Center" />
     <Setter Property="VerticalContentAlignment" Value="Center" />
     <Setter Property="UseSystemFocusVisuals" Value="{StaticResource UseSystemFocusVisuals}" />
-    <contract7Present:Setter Property="CornerRadius" Value="{StaticResource ControlCornerRadius}" />
+    <Setter Property="CornerRadius" Value="{StaticResource ControlCornerRadius}" />
```
[the remark this answers is not in the document]
Comment (TopProgrammer77) [reply src/Uno.UI.FluentTheme.v2/themeresources_v2.xaml]: I am sorry. it's my fault in git work.

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs [hunk 2]
```diff
@@ -112,7 +110,6 @@ private static void ClearContainerForDragDrop(UIElement itemContainer)
 
 			itemContainer.CanDrag = false;
 			itemContainer.DragStarting -= OnItemContainerDragStarting;
-			itemContainer.DropCompleted -= OnItemContainerDragCompleted;
```
[the remark this answers is not in the document]
Comment (TopProgrammer77) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs:115]: https://github.com/unoplatform/uno/blob/72dfb91446cc3a4ca95dea6398665d43bf258596/src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewList.cs#L159

https://github.com/unoplatform/uno/blob/72dfb91446cc3a4ca95dea6398665d43bf258596/src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewList.cs#L178C4-L178C31

Here are the part where the [ClearContainerForDragDrop](https://github.com/unoplatform/uno/blob/72dfb91446cc3a4ca95dea6398665d43bf258596/src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs#L106) function is called.

**Comment by DierkDroth** [comment]
@dr1rrb @MartinZikmund is there still anything missing to get this PR reviewed? 

The issue is getting in the way of my app massively and it would be great to see it being resolved. Thanks in advance

**Comment by DierkDroth** [comment]
@jeromelaban could this please be looked into as well? Please let me know if you felt there is something missing. Thanks

**Review by dr1rrb — APPROVED** [review]

**Comment by jeromelaban** [comment]
@TopProgrammer77 my apologies, I pushed a commit with a typo in it, to fix a winui conversion issue (needed after #14608). Would you mind changing the commit text? Thanks!

**Comment by DierkDroth** [comment]
@TopProgrammer77 @jeromelaban @dr1rrb thanks for working on this, guys.

**Pull request merged**